extensions/amp-story/1.0/amp-story.js

```diff
@@ -669,6 +675,17 @@ export class AmpStory extends AMP.BaseElement {
       ) {
         this.systemLayer_.updateProgress(pageId, progress);
       }
+
+      // We can reliably send 'storyPreviewFinished' from the `PAGE_PROGRESS`
+      // event's callback because story pages autoplay in preview mode.
+      const lastPageId = this.pages_[this.pages_.length - 1]?.element.id;
+      const isLastPage = pageId === lastPageId;
+      const isPreviewFinished =
+        this.getAmpDoc().isPreview() && isLastPage && progress === 1.0;
```

Comment:
Is this correct, or do we only show 20% of the story pages? I got a bit confused wth the logic and missed the latest

Reply:
I've updated the code with params allowing for a configurable `minimum # of pages` and a configurable `% of story that should be shown`. When these params are absent, their default values are 1 page and 30%.